### PR TITLE
replace serviceAccountName key, rbac and cleanup in the code

### PR DESCRIPTION
**Pull request by humblec**
This PR address the below:

- [x] Removal of endpoint based leader election RBAC as its no longer needed
- [x] Replacing `serviceAccount` with `ServiceAccountName` as the former has been deprecated already
- [x] Cleanup in various source files for variable declaration...etc.  

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

/kind cleanup

-->
```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 1]
```diff
@@ -118,4 +118,4 @@ spec:
               mountPath: /csi
       volumes:
         - name: socket-dir
-          emptyDir: {}
+          emptyDir: { }
```
Comment (xing-yang) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:121]: Is this change needed?

**Comment by xing-yang** [comment]
You added commit messages for each commit. Please also add those to the PR description so we know what the changes are for.

File: client/clientset/versioned/fake/register.go [hunk 2]
```diff
@@ -28,8 +28,10 @@ import (
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 )
 
-var scheme = runtime.NewScheme()
-var codecs = serializer.NewCodecFactory(scheme)
+var (
+	scheme = runtime.NewScheme()
+	codecs = serializer.NewCodecFactory(scheme)
+)
```
Comment (xing-yang) [inline client/clientset/versioned/fake/register.go:34]: The files under clientset are automatically generated.  So they should not be manually changed.

File: vendor/github.com/kubernetes-csi/external-snapshotter/client/v6/clientset/versioned/fake/register.go [hunk 3]
```diff
@@ -28,8 +28,10 @@ import (
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 )
 
-var scheme = runtime.NewScheme()
-var codecs = serializer.NewCodecFactory(scheme)
+var (
+	scheme = runtime.NewScheme()
+	codecs = serializer.NewCodecFactory(scheme)
+)
```
Comment (xing-yang) [inline vendor/github.com/kubernetes-csi/external-snapshotter/client/v6/clientset/versioned/fake/register.go:34]: Same here.  Files under clienset are automatically generated.  They should not be changed manually.

**Comment by humblec** [comment]
@xing-yang addressed the comments and also updated the PR description.. ptal. Thanks!

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 4]
```diff
@@ -118,4 +118,4 @@ spec:
               mountPath: /csi
       volumes:
         - name: socket-dir
-          emptyDir: { }
+          emptyDir: {}
```
Comment (xing-yang) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:121]: I thought this change is removed?
